pkg/controller/wip/controller_test.go

```diff
@@ -185,6 +260,248 @@ func TestReconcileInstanceHappyLand(t *testing.T) {
 	}
 
+	// Verify parameters are what we'd expect them to be, basically name, map with two values in it.
+	if len(updateObject.Spec.Parameters.Raw) == 0 {
+		t.Fatalf("Parameters was unexpectedly empty")
+	}
+	if si, ok := fakeInstanceClient.Instances[""]; !ok {
+		t.Fatalf("Did not find the created Instance in fakeInstanceClient after creation")
+	} else {
+		if len(si.Parameters) == 0 {
+			t.Fatalf("Expected parameters but got none")
+		}
+		if e, a := "test-param", si.Parameters["name"].(string); e != a {
+			t.Fatalf("Unexpected name for parameters: expected %v, got %v", e, a)
+		}
+		argsMap := si.Parameters["args"].(map[string]interface{})
+		if e, a := "first-arg", argsMap["first"].(string); e != a {
+			t.Fatalf("Unexpected value in parameter map: expected %v, got %v", e, a)
+		}
+		if e, a := "second-arg", argsMap["second"].(string); e != a {
+			t.Fatalf("Unexpected value in parameter map: expected %v, got %v", e, a)
+		}
+	}
+
+	close(stopCh)
+}
+
+func TestReconcileInstance(t *testing.T) {
+	fakeKubeClient, fakeCatalogClient, fakeBrokerCatalog, fakeInstanceClient, _, testController, sharedInformers, stopCh := newTestController(t)
+
+	fakeBrokerCatalog.RetCatalog = testCatalog
+
+	sharedInformers.Brokers().Informer().GetStore().Add(broker)
+	sharedInformers.ServiceClasses().Informer().GetStore().Add(serviceClass)
+
+	instance := &v1alpha1.Instance{
+		ObjectMeta: v1.ObjectMeta{Name: "test-instance", Namespace: "test-ns"},
+		Spec: v1alpha1.InstanceSpec{
+			ServiceClassName: "test-serviceclass",
+			PlanName:         "default",
+		},
+	}
+
+	testController.reconcileInstance(instance)
+
+	actions := filterActions(fakeCatalogClient.Actions())
+	if e, a := 1, len(actions); e != a {
+		t.Logf("%+v\n", actions)
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
+	}
+
+	// verify no kube resources created
+	kubeActions := fakeKubeClient.Actions()
+	if e, a := 0, len(kubeActions); e != a {
+		t.Fatalf("Unexpected number of actions: expected %v, got %v", e, a)
+	}
+
+	updateAction := actions[0].(core.UpdateAction)
+	if e, a := "update", updateAction.GetVerb(); e != a {
+		t.Fatalf("Unexpected verb on actions[1]; expected %v, got %v", e, a)
+	}
+
+	updateObject := updateAction.GetObject().(*v1alpha1.Instance)
+	if e, a := instance.Name, updateObject.Name; e != a {
+		t.Fatalf("Unexpected name of serviceClass created: expected %v, got %v", e, a)
+	}
+
+	if e, a := 1, len(updateObject.Status.Conditions); e != a {
+		t.Fatalf("Unexpected number of status conditions: expected %v, got %v", e, a)
+	}
+
+	if e, a := v1alpha1.InstanceConditionReady, updateObject.Status.Conditions[0].Type; e != a {
+		t.Fatalf("Unexpected condition type: expected %v, got %v", e, a)
+	}
+
+	if e, a := v1alpha1.ConditionTrue, updateObject.Status.Conditions[0].Status; e != a {
+		t.Fatalf("Unexpected condition status: expected %v, got %v", e, a)
+	}
+
+	// TODO(vaikas): Since the Instance OSBGUID is not set, it gets marked as "" which is certainly
```

Comment:
Ok, so who's job is it to actually fill in the GUID, etc.? Isn't that the job of the controller?